### PR TITLE
refactor: [M3-7385]- MUI v5 Migration - `SRC > Features > Support`

### PR DESCRIPTION
**Pull request by bnussman-akamai**
## Description 📝

Performs general refactors on `SRC > Features > Support` to reduce tech debt, use MUI v5 features, and make Support tickets look _slightly_ less broken.

## Changes  🔄
- Rewrites some class components to functional components
- Makes the UI look a bit less broken
- Rewrite a few enzyme unit tests
- Runs the tss coded on 8 files

## Preview 📷

> [!note]
> Minimal UI changes are expected. Some styles may look different in attempts to make things look better.

| Before  | After   |
| ------- | ------- |
| Screenshot 2023-11-07 at 12 46 44 PM | Screenshot 2023-11-07 at 12 45 39 PM |

## How to test 🧪

> [!warning]
> I recommend testing against the dev environment so we don't create production support tickets

### Verification steps 
- Please check all features under http://localhost:3000/support/tickets
- Verify you can create a support ticket
- Verify you can reply to a support ticket
- Verify you can attach files to a support ticket

## As an Author I have considered 🤔

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a changeset
- [x] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [x] 🚩 Using a feature flag to protect the release
- [x] 👣 Providing comprehensive reproduction steps
- [x] 📑 Providing or updating our documentation
- [x] 🕛 Scheduling a pair reviewing session
- [x] 📱 Providing mobile support
- [x] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/features/Support/AttachFileListItem.tsx [hunk 1]
```diff
@@ -101,12 +96,3 @@ export const AttachFileListItem: React.FC<CombinedProps> = (props) => {
...
   );
 };
-
-const enhanced = compose<CombinedProps, Props>(
-  withHandlers({
-    onClick: (props: Props) => () => {
-      props.removeFile(props.fileIdx);
-    },
-  })
-)(AttachFileListItem);
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Support/AttachFileListItem.tsx:111]: This is nuts 🌰

File: packages/manager/src/features/Support/SupportTicketDetail/TabbedReply/TabbedReply.tsx [hunk 2]
```diff
@@ -59,17 +56,3 @@ const TabbedReply: React.FC<CombinedProps> = (props) => {
     />
   );
 };
-
-/** only update on error and value change */
-const memoized = (component: React.FC<CombinedProps>) =>
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Support/SupportTicketDetail/TabbedReply/TabbedReply.tsx:64]: Over-optimization

File: packages/manager/src/features/Support/SupportTicketDetail/TabbedReply/TicketReply.tsx [hunk 3]
```diff
@@ -1,52 +1,31 @@
...
-type ClassNames = 'replyField';
-
-const styles = () =>
-  createStyles({
-    replyField: {
-      '& > div': {
-        maxWidth: '100% !important',
-      },
-      marginTop: 0,
-    },
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Support/SupportTicketDetail/TabbedReply/TicketReply.tsx:15]: Styles are not needed for this. We can just use the `expand` prop

File: packages/manager/src/features/Support/SupportTickets/SupportTicketDialog.tsx [hunk 4]
```diff
@@ -661,6 +661,7 @@ export const SupportTicketDialog = (props: SupportTicketDialogProps) => {
               label: 'Cancel',
               onClick: onCancel,
             }}
+            sx={{ display: 'flex', justifyContent: 'flex-end' }}
```
Comment (bnussman-akamai) [inline packages/manager/src/features/Support/SupportTickets/SupportTicketDialog.tsx:664]: | before | after |
|--------|-------|
|   Screenshot 2023-11-07 at 1 12 52 PM     |    Screenshot 2023-11-07 at 1 12 45 PM   |

File: packages/manager/src/features/Support/AttachFileListItem.tsx (hunk 1 above)
Comment (jaalah-akamai) [reply packages/manager/src/features/Support/AttachFileListItem.tsx:111]: oh my... 🤯

**Review by jaalah-akamai — APPROVED** [review]
Nice improvements

File: packages/manager/src/features/Support/AttachFileForm.test.tsx [hunk 5]
```diff
@@ -1,25 +1,19 @@
...
 
 import { AttachFileForm } from './AttachFileForm';
 
 const props = {
-  classes: {
-    attachFileButton: '',
-  },
-  files: [attachment1, attachment2],
-  handleFileSelected: jest.fn(),
+  files: [attachment3],
```
Comment (mjac0bs) [inline packages/manager/src/features/Support/AttachFileForm.test.tsx:9]: I'm not familiar with this test but: what happened to attachment1 or attachment2?

**Review by mjac0bs — COMMENTED** [review]
Thanks for the improvements! Between this and the breadcrumb fix, Support has gotten a little facelift recently. 🎉 

I tested opening, updating, and then viewing a closed support ticket in the dev environment. I also tested and confirmed that our special SMTP ticket has no unanticipated UI changes.

I do think we should consider hiding the profile icons in mobile view. The UI looks squashed. This is probably an existing issue, but would be a nice to have here.

Screenshot 2023-11-08 at 1 46 44 PM

File: packages/manager/src/features/Support/AttachFileForm.test.tsx (hunk 5 above)
Comment (bnussman-akamai) [reply packages/manager/src/features/Support/AttachFileForm.test.tsx:9]: I tried making more assertions in the test but they didn't work

**Comment by bnussman-akamai** [comment]
I agree with those changes? Can I do them in a follow up? @mjac0bs

**Comment by mjac0bs** [comment]
> I agree with those changes? Can I do them in a follow up? @mjac0bs

@bnussman-akamai Yep. Just finished looking through the code and will approve in a sec. 🚢

**Review by mjac0bs — APPROVED** [review]

**Pull request merged**